packages/dd-trace/src/config.js

```diff
@@ -181,7 +181,7 @@ function validateNamingVersion (versionString) {
  * @param {string | string[]} input
  */
 function splitJSONPathRules (input) {
-  if (!input) return
+  if (!input || input === '$') return
```

Comment:
Note: this could also be considered a breaking change. I believe with the current state, we can just also handle it as a bugfix.